transform/flatten_mangler.go

```diff
@@ -271,3 +288,41 @@ func getUnderlyingKindType(t reflect.Type) (reflect.Kind, reflect.Type) {
 	}
 	return k, t
 }
+
+func stripPtrs(val reflect.Value) reflect.Value {
+	for val.IsValid() {
+		switch val.Kind() {
+		case reflect.Ptr, reflect.Interface:
+			val = val.Elem()
+		default:
+			return val
+		}
+	}
+	return val
+}
+
+// GetField takes in a reflect.Value and returns the value of the field at the
+// given field path or the zero value if it's not populated
```

Comment:
I think this docstring needs to explain what `t` is used for a bit more explicitly.
Since this is exported, it's part of the public API.

Alternatively, it might be nice to just pass the `reflect.StructField` directly and let the function split it up and pick the right tag. (then we don't need the fieldpath tag exported)

Reply:
`reflect.StructField` would be much better and remove a lot of logic from the flags package. Changed it. Also added a detailed doc string of what the function is doing and the expected output.